loopy/target/pyopencl.py

```diff
@@ -370,6 +370,38 @@ def map_power(self, expr, type_context):
 # }}}
 
 
+# {{{ filter out repeated base storages in IDIS
+
+def convert_global_temp_args_to_base_storage(kernel, idis):
+    from loopy.kernel.data import (InameArg,
+                                   TemporaryVariable)
+    new_idis = []
+
+    for idi in idis:
+        if (idi.offset_for_name is not None
+                or idi.stride_for_name_and_axis is not None
```

Comment:
These should be `ValueArg`s, right? Wouldn't the `isinstance(var_descr, TemporaryVariable)` below suffice to weed them out?